namerd/iface/interpreter-thrift/src/main/scala/io/buoyant/namerd/iface/ThriftNamerInterface.scala

```diff
@@ -265,22 +257,13 @@ class ThriftNamerInterface(
     }
   }
 
-  private[this] val bindingCacheMu = new {}
-  private[this] var bindingCache: Map[(Ns, Dtab, Path), BindingObserver] = Map.empty
-  private[this] def observeBind(ns: String, dtab: Dtab, path: Path): BindingObserver =
-    bindingCacheMu.synchronized {
-      val key = (ns, dtab, path)
-      bindingCache.get(key) match {
-        case Some(obs) =>
-          Trace.recordBinary("namerd.srv/bind.cached", true)
-          obs
-        case None =>
-          Trace.recordBinary("namerd.srv/bind.cached", false)
-          val obs = BindingObserver(interpreters(ns).bind(dtab, path), stamper)
-          bindingCache += (key -> obs)
-          obs
-      }
-    }
+  private[this] val bindingCache = new ObserverCache[(String, Dtab, Path), NameTree[Name.Bound]](
+    activeCapacity = 100,
+    inactiveCapacity = 10
+  )({
+    case (ns, dtab, path) =>
+      BindingObserver(interpreters(ns).bind(dtab, path), stamper)
+  })
```

Comment:
tioli I'd probably write this as:
```scala
{ k =>
  val (ns, dtab, path) = k
  BindingObserver(interpreters(ns).bind(dtab, path), stamper)
}
```

since this makes it clearer that this isn't a partial function...


Also, I'd be inclined to split this out into a separate function so it's clear:

```scala
private[this] case class BindingKey(ns: Ns, dtab: Dtab, name: path)
def mkBindingObserver(k: BindingKey) =
  BindingObserver(...)

private[this] val bindingCache = new ObserverCache[BindingKey, NameTree[Name.Bound]](
  activeCapacity = 100,
  inactiveCapacity = 10,
  mkBindingObserver
)
```